### ISSUE TITLE
LambertConformal

### ISSUE THREAD

**Issue by marqh** [SciTools/cartopy#597]
Hello 

the change
https://github.com/SciTools/cartopy/commit/2a982e942d769a81b7575d466590c1b45543f0d9

has caused problems with the extents of LambertConformal projections which has lead to issues with code dependent on cartopy.

I do not really understand the change, so I'm wary of leaping in and reverting it, but the results I get when using this projection are proving impossible to use.

before:
```
import cartopy.crs as ccrs
import matplotlib.pyplot as plt
ax = plt.axes(projection=ccrs.LambertConformal())
ax.projection._x_limits
(-17567847.747323327, 17567847.747323327)
ax.projection._y_limits
(-9740400.83870995, 14831772.17472166)
```

after:
```
import cartopy.crs as ccrs
import matplotlib.pyplot as plt
ax = plt.axes(projection=ccrs.LambertConformal())
ax.projection._x_limits
(0.0, 3.995460734861362e+252)
ax.projection._y_limits
(0.0, 1.3808377520008518e+267)
```

**Comment by marqh** [comment]
@QuLogic do you have any thoughts on this issue?

thank you
mark

**Comment by QuLogic** [comment]
I don't understand why you get that result. When I run it, I get:
```python
>>> import cartopy.crs as ccrs
>>> import matplotlib.pyplot as plt
>>> ax = plt.axes(projection=ccrs.LambertConformal())
>>> ax.projection._x_limits
(-17567847.747323327, 17567847.747323327)
>>> ax.projection._y_limits
(-9740400.83870995, 14831772.17472166)
```

**Comment by rhattersley** [comment]
I helped @marqh track this down earlier ... it looks to be a problem with shapely. In an earlier version (somewhere around 1.2 if I remember correctly) it couldn't cope with a negatively strided numpy array. It seemed to be ignoring the strides. Hence the [::-1] in the linked commit causing problems. With the bugged version you get things like:
```python
>>> print LineString(np.zeros((4, 3)))
LINESTRING Z (0 0 0, 0 0 0, 0 0 0, 0 0 0)
>>> print LineString(np.zeros((4, 3))[::-1])
LINESTRING Z (0 0 0, 1.635036724600033e+45 6.922162574222758e-310 2.420921664622108e-322, 2.153240850230487e-316 6.922233387896548e-310 6.922374669286862e-310, 1.021226016565964e-312 4.743030200075967e-322 2.371515100037983e-322)
```

I believe @marqh is going to raise a PR which updates the minimum required shapely version. (And perhaps add a warning/error for crufty versions?)

**Comment by QuLogic** [comment]
We could then do `[::-1].copy()` I think, so that it's really reversed instead of a view?

**Comment by marqh** [comment]
> We could then do [::-1].copy() I think, so that it's really reversed instead of a view?

I don't think we should do this, it shouldn't be necessary, the problem is with shapely, not this logic.

I've raised a PR to address this

**Comment by QuLogic** [comment]
This issue was fixed with #600.
